plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/decoders/Decoder.java

```diff
@@ -16,9 +16,10 @@
 import io.trino.spi.block.BlockBuilder;
 import org.opensearch.search.SearchHit;
 
+import java.util.List;
 import java.util.function.Supplier;
 
 public interface Decoder
 {
-    void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output);
+    void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output, List<String> dereferencePath);
```

Comment:
The decoder interface should not have to change. The decoder should be invoked on the final element after the dereference chain encompassing the pushed down expression is resolved.

Reply:
Agreed. Removed decoder interface changes.